components/cldr-json-data-provider/src/cldr_paths.rs

```diff
@@ -1,21 +1,50 @@
-use crate::error::MissingSourceError;
+use crate::error::{Error, MissingSourceError};
 use std::default::Default;
 use std::path::PathBuf;
 
-/// Struct containing filesystem paths to the CLDR JSON resource directories.
+/// Trait returning filesystem paths to CLDR JSON resource directories.
 /// The fields should be Ok if present. They default to Err when not present.
+pub trait CldrPaths {
+    fn cldr_core(&self) -> Result<PathBuf, Error>;
+    fn cldr_dates(&self) -> Result<PathBuf, Error>;
```

Comment:
do you need to return `PathBuf`? Usually in such cases we use `Path` (just like we'd return `&str` not `String`).

Reply:
I think you either have an owned `PathBuf` or a borrowed `&Path`.  If I return `&Path`, then someone has to own it.

I figured it was fine to return `PathBuf` because the only call site immediately takes it and starts appending more pieces to it, so if we returned `&Path`, we'd need to call `.to_path_buf()` anyway.